hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -209,9 +224,16 @@ public IOException getIoException() {
   @Override
   public void write(int b) throws IOException {
     checkOpen();
-    byte[] buf = new byte[1];
-    buf[0] = (byte) b;
-    write(buf, 0, 1);
+    if (currentBufferRemaining == 0) {
+      allocateNewBuffer();
+    }
+    currentBuffer.put((byte) b);
+    currentBufferRemaining--;
+    if (currentBufferRemaining == 0) {
+      writeChunk(currentBuffer);
+    }
```

Comment:
For consistency, I think we should move the condition to a new method `writeChunkIfNeeded`.

Reply:
Not sure if I understood. Why is it more consistent to move this 3 lines to a separated method?

Reply:
`checkOpen()` and `doFlushOrWatchIfNeeded()` calls are both concise.  I think moving the conditions for these two other calls would help make the level of abstraction in `write()` more similar.  But it's optional.

Reply:
Ok, I think I got the idea, but please double-check my commit as I am not sure ;-)

9ab01a7753b75009997cc33a338bbef282230687

Reply:
Thanks for the update.